### ISSUE TITLE
Support for STM32F4

### ISSUE THREAD

**Issue by ipsod** [gin66/FastAccelStepper#224]
A lot of the new 8-axis printer boards are running STM32F4 chips.

I understand if there isn't time, I just wanted to put this on the map.

**Comment by gin66** [comment]
good proposal. PR‘s are welcome

**Comment by ajay01994** [comment]
Hi ,
I see the teensy step doesnt support teensy 4.0 and above, do you have any reference implementation

**Comment by RoboCrafty** [comment]
@ipsod Found any way to use fastaccelstepper on the stm32f4 boards? I really want the addtoqueue functionality for using cuwstom motion profiles which teensystep lacks

**Comment by gambledusty** [comment]
@RoboCrafty No, I ended up using TMC stepper drivers' built-in speed control.  They don't allow the same level of step-perfect control, so I kind of close the loop on the MCU with a PID, same as you might for a servo, but instead of reading encoder position, you just read what the TMC driver reports.

I'd still like to get this figured out, though.  Might be easier now, with AI to help.

**Comment by RoboCrafty** [comment]
@ipsod Interesting! Yeah i read about their six point ramp motion control. No idea if there are any libs to interact with that. Are you using TMC5160? So you only send position via SPI and have left the step/dir pins unconnected?

I am now considering just switching to TeensyStep and getting the BTT Octopus, afaik it should work with TeensyStep as it uses STM32F446ZET6 which comes under the supported chips, any thoughts on that?

I also came across a thread of TeensyStep git issues where they discuss using custom motion profiles with that library

**Comment by gambledusty** [comment]
@RoboCrafty No, I'm not using their ramping, just speed control, which works even on TMC2209, and definitely on all the more advanced ones I think.  Right, no step/dir pins, just SPI or UART, depending on the driver.  I've got a control loop running on the microcontroller that can calculate what position and speed should be at any moment, according to the plan, but rather than feeding that directly, it goes to a PID, so that I can use the positional error for path correction (small errors happen due to float-math and timing).  It's kind of a strange way to control steppers, but it also makes it easy to swap out for any other type of motor+encoder combination.

I don't know about TeensyStep - seems like it should work, but I never ended up using it.

**Comment by RoboCrafty** [comment]
> [@RoboCrafty](https://github.com/RoboCrafty) No, I'm not using their ramping, just speed control, which works even on TMC2209, and definitely on all the more advanced ones I think. Right, no step/dir pins, just SPI or UART, depending on the driver. I've got a control loop running on the microcontroller that can calculate what position and speed should be at any moment, according to the plan, but rather than feeding that directly, it goes to a PID, so that I can use the positional error for path correction (small errors happen due to float-math and timing). It's kind of a strange way to control steppers, but it also makes it easy to swap out for any other type of motor+encoder combination.
> 
> I don't know about TeensyStep - seems like it should work, but I never ended up using it.

Interesting! I cant seem to find anything about controlling tmc2209 without step/dir on the datasheet, could you tell me which section talks about this? I thought the uart on tmc2209 is just for configs

**Comment by gambledusty** [comment]
https://github.com/janelia-arduino/TMC2209

search for "velocity"
